src/components/KTable/KTable.vue

```diff
@@ -106,26 +105,42 @@
                   sortClickHandler(column)
                 }
               }"
+              @mouseleave="currentFocusedColumn = ''"
+              @mouseover="currentFocusedColumn = column.key"
             >
-              <span class="k-table-headers-container">
-                <slot
-                  :column="getGeneric(column)"
-                  :name="getColumnSlotName(column.key)"
-                >
-                  <span :class="{'sr-only': column.hideLabel}">
-                    {{ column.label ? column.label : column.key }}
-                  </span>
-                </slot>
-
-                <KIcon
-                  v-if="!disableSorting && !column.hideLabel && column.sortable"
-                  aria-hidden="true"
-                  class="caret"
-                  :color="`var(--kui-color-text, ${KUI_COLOR_TEXT})`"
-                  icon="chevronDown"
-                  :size="KUI_ICON_SIZE_20"
-                />
-              </span>
+              <div
+                :aria-sort="!disableSorting && column.key === sortColumnKey ? (sortColumnOrder === 'asc' ? 'ascending' : 'descending') : undefined"
+                class="k-table-headers"
+                :class="{
+                  'sortable': !disableSorting && !column.hideLabel && column.sortable,
+                  'active-sort': !disableSorting && !column.hideLabel && column.sortable && column.key === sortColumnKey,
+                  [sortColumnOrder]: !disableSorting && column.key === sortColumnKey && !column.hideLabel,
+                  'is-scrolled': isScrolled,
+                  'truncated-column resizable': resizeColumns
+                }"
+                :data-testid="`k-table-header-${column.key}`"
+                :style="widthStyles[column.key]"
```

Comment:
the inner `<div>` only gets the `width` style set so the `resize` handle can actually move - this will probably be able to go away